src/entities/ShareClass.ts

```diff
@@ -1666,4 +1673,65 @@ export class ShareClass extends Entity {
       )
     )
   }
+
+  /** @internal */
+  _tokenInstancePositions() {
+    return this._query(null, () =>
+      combineLatest([
+        this._root._deployments(),
+        this.pool.currency(),
+        this.vaults(),
+        this._root._queryIndexer<{
+          tokenInstancePositions: {
+            items: {
+              accountAddress: HexString
+              centrifugeId: string
+              balance: bigint
+              isFrozen: boolean
+            }[]
+          }
+        }>(
+          `query ($scId: String!) {
+          tokenInstancePositions(where: { tokenId: $scId }) {
+            items {
+              accountAddress
+              centrifugeId
+              balance
+              isFrozen
+            }
+          }
+        }`,
+          { scId: this.id.raw }
+        ),
+      ]).pipe(
+        switchMap(([deployments, poolCurrency, vaults, { tokenInstancePositions }]) => {
+          const chainsById = new Map(deployments.blockchains.items.map((chain) => [chain.centrifugeId, chain.id]))
+
+          return combineLatest(
+            tokenInstancePositions.items.map((position) => {
+              const chainId = Number(chainsById.get(position.centrifugeId)!)
+              const vault = vaults.find((v) => v.chainId === chainId)
```

Comment:
getting a vault here doesn't make sense. they can have invested in multiple vaults or none at all. the only relevant number is the balance of share tokens